bin/lib/ce_install.py

```diff
@@ -66,256 +90,324 @@ def squash_mount_check(rootfolder, subdir, context):
+    libyaml.add_top_rust_crates()
+    libyaml.save()
+
+
+@cli.command()
+@click.pass_obj
+def generate_rust_props(context: CliContext):
+    """Generate Rust property files for crates."""
+    propfile = Path(os.path.join(os.curdir, 'props'))
+    with propfile.open(mode="w", encoding="utf-8") as file:
+        libyaml = LibraryYaml(context.installation_context.yaml_dir)
+        props = libyaml.get_ce_properties_for_rust_libraries()
+        file.write(props)
+
+
+@cli.command()
+@click.pass_obj
+@click.argument("libid")
+@click.argument("libversion")
+def add_crate(context: CliContext, libid: str, libversion: str):
+    """Add crate LIBID version LIBVERSION."""
+    libyaml = LibraryYaml(context.installation_context.yaml_dir)
+    libyaml.add_rust_crate(libid, libversion)
+    libyaml.save()
 
-            if args.force or installable.should_build():
-                if not installable.is_installed():
-                    _LOGGER.info("%s is not installed, unable to build", installable.name)
-                    num_skipped += 1
-                else:
-                    try:
-                        [num_installed, num_skipped, num_failed] = installable.build(args.buildfor)
-                        if num_installed > 0:
-                            _LOGGER.info("%s built OK", installable.name)
-                        elif num_failed:
-                            _LOGGER.info("%s failed to build", installable.name)
-                    except RuntimeError as e:
-                        if args.buildfor:
-                            raise e
-                        else:
-                            _LOGGER.info("%s failed to build: %s", installable.name, e)
-                            num_failed += 1
-            else:
-                _LOGGER.info("%s is already built, skipping", installable.name)
-                num_skipped += 1
-        print(f'{num_installed} packages built OK, {num_skipped} skipped, and {num_failed} failed build')
-        if num_failed:
-            sys.exit(1)
-        sys.exit(0)
-    elif args.command == 'reformat':
-        libyaml = LibraryYaml(args.yaml_dir)
-        libyaml.reformat()
-    elif args.command == 'addtoprustcrates':
-        libyaml = LibraryYaml(args.yaml_dir)
-        libyaml.add_top_rust_crates()
-        libyaml.save()
-    elif args.command == 'generaterustprops':
-        propfile = Path(os.path.join(os.curdir, 'props'))
-        with propfile.open(mode="w", encoding="utf-8") as file:
-            libyaml = LibraryYaml(args.yaml_dir)
-            props = libyaml.get_ce_properties_for_rust_libraries()
-            file.write(props)
-    elif args.command == 'addcrate':
-        libyaml = LibraryYaml(args.yaml_dir)
-        libyaml.add_rust_crate(args.filter[0], args.filter[1])
-        libyaml.save()
-
-    else:
-        raise RuntimeError("Er, whoops")
+
+def main():
+    cli(prog_name='ce_install')  # pylint: disable=unexpected-keyword-arg,no-value-for-parameter
```

Comment:
you can add some lines to pylintrc to make this do the correct thing: https://stackoverflow.com/a/72212721

Reply:
thanks!